x/dispensation/types/records.go

```diff
@@ -1,42 +1,12 @@
 package types
 
 import (
-	"fmt"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"strings"
 )
 
 //This package is used to keep historical data. This will later be used to distribute rewards over different blocks through a gov proposal
 
-type ClaimStatus int64
-
-const Pending ClaimStatus = 1
-const Completed ClaimStatus = 2
-
-func (d ClaimStatus) String() string {
-	switch d {
-	case Pending:
-		return "Pending"
-	case Completed:
-		return "Completed"
-	default:
-		return "All"
-	}
-}
-
-// DistributionRecord is created for every recipient for a distribution
-// TODO : Add ClaimStatus to the prefixed key for records
-type DistributionRecord struct {
-	ClaimStatus
-	DistributionName            string         `json:"distribution_name"`
-	RecipientAddress            sdk.AccAddress `json:"recipient_address"`
-	Coins                       sdk.Coins      `json:"coins"`
-	DistributionStartHeight     int64          `json:"distribution_start_height"`
-	DistributionCompletedHeight int64          `json:"distribution_completed_height"`
-}
-type DistributionRecords []DistributionRecord
-
-func NewDistributionRecord(distributionName string, recipientAddress sdk.AccAddress, coins sdk.Coins, start int64, end int64) DistributionRecord {
+func NewDistributionRecord(distributionName string, recipientAddress string, coins sdk.Coins, start sdk.Int, end sdk.Int) DistributionRecord {
```

Comment:
Any reason why we need sdk.Int for start and end heights ?  THe block height is still int64 right ?

Reply:
JUst checked this has already been pointd out by marko , keeping this comment to help make the required changes